app/models/base_input.rb

```diff
@@ -0,0 +1,39 @@
+class BaseInput
```

Comment:
> It seems like both classes need to share some implementation (hence a module) but [cannot be substituted](https://en.wikipedia.org/wiki/Liskov_substitution_principle) for one another.

The LSP does not apply here, it is about subtypes, not "sibling-types" :wink:
But I agree that the base class does not feel like a class - e.g. it does not have any data in it. Hence, I am in favour of making this a module. :+1: